src/telescope_control.py

```diff
@@ -683,12 +683,19 @@ def run_track(ctx: _RunContext, conn: mpc.Connection):
                     ]
                 )
 
-                if not _ag_wait_one_group(activity, activity_groups):
-                    if activity_groups[0] == intercept_group:
-                        ctx.log.info("canceled while intercepting")
-                    else:
-                        ctx.log.info("canceled while tracking")
-                    break
+                match _ag_wait_one_group(activity, activity_groups):
+                    case None:
+                        if activity_groups[0] == intercept_group:
+                            ctx.log.info("canceled while intercepting")
+                        else:
+                            ctx.log.info("canceled while tracking")
+                        break
+                    case completed if completed is intercept_group:
+                        with activity._cond:
+                            status = activity._status
+                        conn.send(_RequestProgress(goal, status, "slew_complete"))
+                    case _:
+                        pass
```

Comment:
This is where the real magic happens.  We used to just wait for cancellation, but now `_ag_wait_one_group` returns the group that it finished waiting on, or `None` if it was interrupted.  So, with that value in hand, we can check if the completed item was the intercept group (slew), and if so, emit a progress message, which makes its way back to the main process.